src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -599,20 +587,29 @@ public void constraintsChanged()
      * {@link ConferenceSpeechActivity}.
      * @param bweBps the current bandwidth estimation (in bps)
      */
-    private synchronized void update(List<AbstractEndpoint> conferenceEndpoints, long bweBps)
+    private synchronized void update()
     {
-        lastEndpointOrdering = conferenceEndpoints;
-        lastBwe = bweBps;
+        long nowMs = System.currentTimeMillis();
+
+        long bweBps = getAvailableBandwidth(nowMs);
 
         logger.debug("TEMP: BitrateController " + hashCode() + " updating. " +
                 "bwebps is " + bweBps);
 
         // Create a copy as we may modify the list in the prioritize method.
-        conferenceEndpoints = new ArrayList<>(conferenceEndpoints);
+        List<AbstractEndpoint> sortedEndpoints = new ArrayList<>(sortedEndpointIds.size());
+        for (String endpointId : sortedEndpointIds)
+        {
+            AbstractEndpoint abstractEndpoint = destinationEndpoint.getConference().getEndpoint(endpointId);
+            if (abstractEndpoint != null)
+            {
+                sortedEndpoints.add(abstractEndpoint);
+            }
+        }
 
         // Compute the bitrate allocation.
         TrackBitrateAllocation[]
-            trackBitrateAllocations = allocate(bweBps, conferenceEndpoints);
+            trackBitrateAllocations = allocate(bweBps, sortedEndpoints);
         logger.debug("TEMP: BitrateController#update got " + trackBitrateAllocations.length +
```

Comment:
maybe wrap this in an `isDebugEnabled` check and remove the TEMP line from it while you're in there?